cpp/CMakeLists.txt

```diff
@@ -1,6 +1,10 @@
 # ------------------------------------------------------------------------------
 # Top level CMakeLists.txt file for DOLFINx
-cmake_minimum_required(VERSION 3.19)
```

Comment:
Let's just require 3.21 - different requirements adds complexity.

Reply:
Breaks Rocky 9 as it only has CMake 3.19.

Reply:
> Breaks Rocky 9 as it only has CMake 3.19.

Seems to me to have 3.26.